bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -534,15 +536,32 @@ private void doRefreshShade(RefreshKind kind) {
                             surveyData.forEach(data -> joiner.add(data.toString()));
                             logger.debug("Survey response for shade {}: {}", shadeId, joiner.toString());
                         }
+
+                        int hubRssi = Integer.MAX_VALUE;
+                        int repeaterRssi = Integer.MAX_VALUE;
+                        for (SurveyData survey : surveyData) {
+                            if (survey.neighborId == 0) {
+                                hubRssi = survey.rssi;
+                            } else {
+                                repeaterRssi = survey.rssi;
+                            }
+                        }
+                        updateState(CHANNEL_SHADE_HUB_RSSI, hubRssi == Integer.MAX_VALUE ? UnDefType.UNDEF
+                                : new QuantityType<>(hubRssi, Units.DECIBEL_MILLIWATTS));
+                        updateState(CHANNEL_SHADE_REPEATER_RSSI, repeaterRssi == Integer.MAX_VALUE ? UnDefType.UNDEF
+                                : new QuantityType<>(repeaterRssi, Units.DECIBEL_MILLIWATTS));
+
                         shadeData = webTargets.getShade(shadeId);
                         updateSignalStrengthState(shadeData.signalStrength);
                     } else {
                         logger.info("No data from shade {} survey", shadeId);
                         /*
-                         * Setting channel to UNDEF here would be reverted on next poll, since
-                         * signal strength is part of shade response. So leaving current value,
+                         * Setting signal strength channel to UNDEF here would be reverted on next poll,
+                         * since signal strength is part of shade response. So leaving current value,
```

Comment:
Looks like the comment is wrong as you are no more leaving the current value.

Reply:
The **signalStrength** channel is still left with current value, but the new RSSI channels are updated to **UNDEF** on purpose. It's a bit tricky: signal strength is part of the **shade** response (which also includes positions, battery level etc.). But in order to force refresh the value, a **survey** request has to be made prior to the **shade** request. This is how the PowerView app works, as I was intercepting traffic from it.

Reply:
Ok